diff --git a/arch/arm/boot/dts/bcm2708-rpi-b-plus.dts b/arch/arm/boot/dts/bcm2708-rpi-b-plus.dts
index 9839960c753f..d221e972cf29 100644
--- a/arch/arm/boot/dts/bcm2708-rpi-b-plus.dts
+++ b/arch/arm/boot/dts/bcm2708-rpi-b-plus.dts
@@ -144,6 +144,27 @@ &i2c1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&i2c1_pins>;
 	clock-frequency = <100000>;
+
+        power-supply@58 {
+                compatible = "lenovo,crps";
+                reg = <0x58>;
+        };
+
+        power-supply@59 {
+                compatible = "lenovo,crps";
+                reg = <0x59>;
+        };
+
+        power-supply@68 {
+                compatible = "lenovo,cff4";
+                reg = <0x68>;
+        };
+
+        power-supply@69 {
+                compatible = "lenovo,cff4";
+                reg = <0x69>;
+        };
+
 };
 
 &i2c2 {
