diff --git a/arch/arm/boot/dts/bcm2835-rpi.dtsi b/arch/arm/boot/dts/bcm2835-rpi.dtsi
index 87ddcad76083..7d370c841c0d 100644
--- a/arch/arm/boot/dts/bcm2835-rpi.dtsi
+++ b/arch/arm/boot/dts/bcm2835-rpi.dtsi
@@ -61,6 +61,37 @@ &i2c1 {
 	pinctrl-0 = <&i2c1_gpio2>;
 	status = "okay";
 	clock-frequency = <100000>;
+        status = "okay";
+
+        power-supply@58 {
+                compatible = "lenovo,crps";
+                reg = <0x58>;
+        };
+
+        power-supply@59 {
+                compatible = "lenovo,crps";
+                reg = <0x59>;
+        };
+
+        power-supply@5a {
+                compatible = "lenovo,crps";
+                reg = <0x5a>;
+        };
+
+        power-supply@5b {
+                compatible = "lenovo,crps";
+                reg = <0x5b>;
+        };
+
+        power-supply@68 {
+                compatible = "lenovo,cff4";
+                reg = <0x68>;
+        };
+
+        power-supply@69 {
+                compatible = "lenovo,cff4";
+                reg = <0x69>;
+        };
 };
 
 &usb {
